### ISSUE TITLE
[JENKINS-6846] Poll SCM with post-commit hook or by cron-definition

### ISSUE THREAD

**Issue by timja-bot** [timja-org/jenkins-gh-issues-poc-11-04#792]
<p>We have splitted some projects from one single job (doing build/test/analyse) to two jobs (one only for build and second one for test/analyse). The first job polls scm every 10 minutes, the second one polls scm at midnight.</p>

<p>Now we defined a svn post-commit hook for CI, so we don't have to poll every 10 minutes for changes. This is fine for the first job, but the second one now starts too (because 'Poll SCM' is checked). For now it is not possible to define a job, that polls scm due a cron-definition and another one that is triggered by post-commit hook.</p>

<p>It would be nice to have an additional option like "ignore post-commit hook". So 'Poll SCM' could be checked and triggered by cron-definition and not by post-commit.</p>

---
<details><summary><i>Originally reported by <a title="manthos" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=manthos">manthos</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-6846" target="_blank">Poll SCM with post-commit hook or by cron-definition</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="kutzi" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kutzi">kutzi</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Critical
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2012-11-24T15:25:00+00:00
<li><b>imported</b>: 2025-11-04
</ul></i>
</details>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kutzi">kutzi</a>:</i>
<p>I support that request. The post commit hook doesn't allow jobs like: 'run some reports every 2 hours, but only if SCM has changed'.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=manthos">manthos</a>:</i>
<p>Changed this improvement to critical.</p>

<p>For now it is not possible to have jobs triggered by SCM postcommithook and jobs triggered only by polling SCM at a defined time. Only one or other mode is possible; hudson wide...</p>

<p>A option like this could bring full flexibility back:</p>

<p>[ ] Poll SCM<br/>
   [ ] by postcommithook<br/>
   [ ] Schedule (with cron)</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kutzi">kutzi</a>:</i>
<p>For easier migration of existing configurations it would e.g. be possible to just add another checkbox<br/>
<span class="error">&#91;x&#93;</span> listen to post-commit-hooks<br/>
which would be true by default (and for migrated configs).<br/>
If you only want to poll by cron you only would have to disable the checkbox</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gbois">gbois</a>:</i>
<p>@manthos<br/>
request information:<br/>
1) Do you have think that the second job could poll build result of job1 instead of SCM<br/>
2) How do you schedule the first job in the post commit hook?</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=manthos">manthos</a>:</i>
<p>1) this doesn't change anything. if I bind job2 to job1, job2 builds right after job1...<br/>
2) first job starts building by a scm postcommithook (push), no cronjob needed.</p>

<p>IMO 'Poll SCM' and 'postcommithook (push)' are different triggers. Currently you are not able to have different Builds (based on same scm repository) but one triggered by postcommithook (push), the other by poll.</p>

<p>Example:<br/>
Job1:</p>
<ul class="alternate" type="square">
	<li>maven deploy -DskipTests=true of <a href="http://svn.foo.bar/svn/java/foobarproject" class="external-link" target="_blank" rel="nofollow noopener">http://svn.foo.bar/svn/java/foobarproject</a></li>
	<li>build triggered by scm postcommithook (push)<br/>
So every time a user checks something into scm, foobarproject builds, skipping the (time consuming) tests</li>
</ul>


<p>Job2:</p>
<ul class="alternate" type="square">
	<li>maven test of <a href="http://svn.foo.bar/svn/java/foobarproject" class="external-link" target="_blank" rel="nofollow noopener">http://svn.foo.bar/svn/java/foobarproject</a></li>
	<li>build triggered by Poll SCM, in ex: (Mo-Fr at 23:00 -> 0 23 * * 1,2,3,4,5)<br/>
So at 23:00 -> tests of foobarproject but only if something checked in (since last run)</li>
</ul>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rainerw">rainerw</a>:</i>
<p>+1 <br/>
Same Problem: running nightly UI tests, which take about 3-4 hours to finish. I want this to run only after an actual change, because running every midnight will make the test-result-history kind of useless.</p>

<p>Therefore this is definitively a critical bug!</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Christoph Kutzinski<br/>
Path:<br/>
 changelog.html<br/>
 core/src/main/java/hudson/triggers/SCMTrigger.java<br/>
 core/src/main/resources/hudson/triggers/SCMTrigger/config.jelly<br/>
 core/src/main/resources/hudson/triggers/SCMTrigger/help-ignorePostCommitHooks.html<br/>
<a href="http://jenkins-ci.org/commit/jenkins/b65c30cb368e79205edacd958d8fd98dd89f3a73" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/b65c30cb368e79205edacd958d8fd98dd89f3a73</a><br/>
Log:<br/>
  Switch to ignore post-commit hook in SCM polling triggers <span class="error">&#91;FIXED JENKINS-6846&#93;</span></p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=scm_issue_link">scm_issue_link</a>:</i>
<p>Code changed in jenkins<br/>
User: Christoph Kutzinski<br/>
Path:<br/>
 src/main/java/hudson/scm/SubversionRepositoryStatus.java<br/>
<a href="http://jenkins-ci.org/commit/subversion-plugin/2de66bbcc17e80933d754bba09d000b4c6d1ab4d" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/subversion-plugin/2de66bbcc17e80933d754bba09d000b4c6d1ab4d</a><br/>
Log:<br/>
  Switch to ignore post-commit hook in SCM polling triggers <span class="error">&#91;FIXED JENKINS-6846&#93;</span></p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dogfood">dogfood</a>:</i>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/yellow.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/2096/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #2096</a><br/>
     Switch to ignore post-commit hook in SCM polling triggers <span class="error">&#91;FIXED JENKINS-6846&#93;</span> (Revision b65c30cb368e79205edacd958d8fd98dd89f3a73)</p>

<p>     Result = UNSTABLE<br/>
Christoph Kutzinski : <a href="https://github.com/jenkinsci/jenkins/commit/b65c30cb368e79205edacd958d8fd98dd89f3a73" class="external-link" target="_blank" rel="nofollow noopener">b65c30cb368e79205edacd958d8fd98dd89f3a73</a><br/>
Files : </p>
<ul>
	<li>core/src/main/resources/hudson/triggers/SCMTrigger/help-ignorePostCommitHooks.html</li>
	<li>changelog.html</li>
	<li>core/src/main/resources/hudson/triggers/SCMTrigger/config.jelly</li>
	<li>core/src/main/java/hudson/triggers/SCMTrigger.java</li>
</ul>
